src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1615,45 +1615,40 @@ const char16_t* Compiler::eeGetShortClassName(CORINFO_CLASS_HANDLE clsHnd)
     return param.classNameWidePtr;
 }
 
-const WCHAR* Compiler::eeGetCPString(size_t strHandle)
+void Compiler::eePrintFrozenObjectDescription(const char* prefix, size_t handle)
 {
-#ifdef HOST_UNIX
-    return nullptr;
-#else
-    char buff[512 + sizeof(CORINFO_String)];
-
-    // make this bulletproof, so it works even if we are wrong.
-    if (ReadProcessMemory(GetCurrentProcess(), (void*)strHandle, buff, 4, nullptr) == 0)
-    {
-        return (nullptr);
-    }
-
-    CORINFO_String* asString = nullptr;
-    if (impGetStringClass() == *((CORINFO_CLASS_HANDLE*)strHandle))
+    const int maxStrSize = 64;
+    char16_t  str[maxStrSize];
+    int       realLength = this->info.compCompHnd->objectToString((void*)handle, str, maxStrSize);
+    if (realLength >= maxStrSize)
     {
-        // strHandle is a frozen string
-        // We assume strHandle is never an "interior" pointer in a frozen string
-        // (jit is not expected to perform such foldings)
-        asString = (CORINFO_String*)strHandle;
+        str[maxStrSize - 4] = L'.';
+        str[maxStrSize - 3] = L'.';
+        str[maxStrSize - 2] = L'.';
+        str[maxStrSize - 1] = 0;
     }
     else
     {
-        // strHandle is a pinned handle to a string object
-        asString = *((CORINFO_String**)strHandle);
+        str[realLength] = 0;
     }
 
-    if (ReadProcessMemory(GetCurrentProcess(), asString, buff, sizeof(buff), nullptr) == 0)
+    for (size_t i = 0; i < min(maxStrSize, realLength); i++)
     {
-        return (nullptr);
+        // Escape \n and \r symbols
+        if (str[i] == L'\n' || str[i] == L'\r')
+        {
+            str[i] = L' ';
+        }
     }
 
-    if (asString->stringLen >= 255 || asString->chars[asString->stringLen] != 0)
+    if (realLength >= 0)
     {
-        return nullptr;
+        printf("%s '%S'\n", prefix, str);
```

Comment:
It would be better to return the string from the JIT/EE interface API as utf8, similar to other 

The problem with depending on `%S` to format `char16_t` strings is that depends on the PAL emulating Windows CRT behavior on Unix. We have been working towards getting rid of the Windows PAL emulator, and adding more uses of %S goes against that.

Reply:
@jkotas I was thinking of leaving it as is just for future JIT optimizations e.g. enable unrolling of Equals/StartsWith for static readonly strings, folding some `.ToString()` operations etc and for that I'll need raw content

Reply:
If you want to do that, it should be a new API that just returns the raw string content.

I do not think it is a good idea to be overloading APIs to do both pretty printing and real functionality.

Reply:
Replaced with UTF8 `char*`
The only problem that it now prints garbage for non latin1 string literals for JitDisasm

Reply:
The same issue exists in number of other places (e.g. for method names). I do not think that it is a problem. Do you agree?

I think we should standardize on utf8 output, and tell everybody who is not happy about the garbage characters from jit tracing to switch their console encoding to utf8. I have my console encoding set to utf8 on all my Windows systems for a while, so I almost forgot that this problem exists.